scripts/lib/CIME/build.py

```diff
@@ -21,6 +21,38 @@
      "CAM_CONFIG_OPTS", "COMP_LND", "COMPARE_TO_NUOPC", "HOMME_TARGET",
      "OCN_SUBMODEL", "CISM_USE_TRILINOS", "USE_TRILINOS", "USE_ALBANY", "USE_PETSC")
 
+def generate_makefile_macro(case, caseroot, comp_name):
+    """
+    Only used by E3SM. Generates a flat Makefile macro file based on the CMake cache system.
+    This macro is only used by certain sharedlibs since components use CMake.
+    Since indirection based on comp_name is allowed for sharedlibs, each sharedlib must generate
+    their own macro.
+    """
+    new_cmake_dir = os.path.join(caseroot, "cmake_macros")
+    if not os.path.isdir(new_cmake_dir):
+        return
+
+    cmake_macro = os.path.join(caseroot, "Macros.cmake")
+    expect(os.path.exists(cmake_macro), "Cannot generate Makefile macro without {}".format(cmake_macro))
+
+    cmake_args = get_standard_cmake_args(case, "DO_NOT_USE", shared_lib=True)
+    output = run_cmd_no_fail("cmake -DCONVERT_TO_MAKE=ON -DCOMP_NAME={} {} -C Macros.cmake |& grep E3SM_SET_MAKEFILE_VAR | grep -v BUILD_INTERNAL_IGNORE".format(comp_name, cmake_args), from_dir=caseroot)
+
+    # The Tools/Makefile may have already adding things to CPPDEFS and SLIBS
+    real_output = output.replace("E3SM_SET_MAKEFILE_VAR ", "").\
+                  replace("CPPDEFS := ", "CPPDEFS := $(CPPDEFS) ").\
+                  replace("SLIBS := ", "SLIBS := $(SLIBS) ")
+
+    with open(os.path.join(caseroot, "Macros.make"), "w") as fd:
```

Comment:
pylint is complaining unless encoding is added to open statements.

Reply:
@jedwards4b , both my pylint and the CI test machine OK'd this file. Can you paste the pylint warning you are getting?